lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -4932,6 +4930,43 @@ namespace Js
         }
     }
 
+    bool JavascriptOperators::CanShortcutOnUnknownPropertyName(RecyclableObject *instance)
+    {
+        if (!CanShortcutInstanceOnUnknownPropertyName(instance))
+        {
+            return false;
+        }
+        return CanShortcutPrototypeChainOnUnknownPropertyName(instance->GetPrototype());
+    }
+
+    bool JavascriptOperators::CanShortcutInstanceOnUnknownPropertyName(RecyclableObject *instance)
+    {
+        if (JavascriptProxy::Is(instance))
+        {
+            return false;
+        }
+        if (!instance->HasDeferredTypeHandler())
+        {
+            return true;
+        }
+        return !(instance->IsExternal() || 
```

Comment:
Ah.  Is it not possible for a DOM object to still have properties that the current undeferred type has not seen though?  Or do we fill all properties once we undefer the type?  Seems like we couldn't guarantee that the type fully matches the DOM object since the host could add new properties after we have undeferred the type, or do I misunderstand external objects?